lib/client/keyagent.go

```diff
@@ -80,12 +80,17 @@ func NewLocalAgent(keyDir string, proxyHost string, username string) (a *LocalKe
 		}),
 		Agent:     agent.NewKeyring(),
 		keyStore:  keystore,
-		sshAgent:  connectToSSHAgent(),
 		noHosts:   make(map[string]bool),
 		username:  username,
 		proxyHost: proxyHost,
 	}
 
+	if !noLocalSSHAgent {
```

Comment:
if not no local agent?

Reply:
I tried to keep the zero-value of `client.Config` usable so that not setting `NoLocalSSHAgent` would preserve existing behavior.
But you're right, this is clunky. Flipped the `Config` field and set it in `MakeDefaultConfig` to `true`.